### PR TITLE
Update undisturbed ground temperatures

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Use an adjustment, `y = 0.91x + 1.82` on `weather.data.AnnualAvgDrybulb`, for ground temperatures. The adjustment is based on a correlation between mean drybulb temperature (C) and ground temperature (C) from L. Xing 2014 model.

Also, in the `gshp_hxbore_ft_per_ton` method for calculating bore length, use the adjustment instead of the annual average drybulb temperature.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [ ] ~Schematron validator (`EPvalidator.xml`) has been updated~
- [ ] ~Sample files have been added/updated (via `tasks.rb`)~
- [x] Tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests` and/or `workflow/tests/hpxml_translator_test.rb`)
- [ ] ~Documentation has been updated~
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:2683]: Both `nom_length_heat` and `nom_length_cool` use an annual average value. Would it be better to use the min and max (or perhaps winter and summer) monthly temperatures, respectively?

File: HPXMLtoOpenStudio/tests/test_weather.rb
Comment (shorowit) [inline HPXMLtoOpenStudio/tests/test_weather.rb:262]: This could be improved. For example, CZ 8 should clearly use the 8AK value. 7 could be the average of 7A/7B unless it's also in AK, in which case it'd use 8AK. Etc.

File: HPXMLtoOpenStudio/resources/data/ground_temperatures.csv
Comment (afontani) [reply HPXMLtoOpenStudio/resources/data/ground_temperatures.csv:1]: **Context of where the values came from:** The data is from L. Xing's 2014 [dissertation](https://shareok.org/bitstream/handle/11244/15208/Xing_okstate_0664D_13659.pdf?sequence=1) that created this ground temperature model in E+ (https://bigladdersoftware.com/epx/docs/8-7/engineering-reference/undisturbed-ground-temperature-model-xing.html). Appendix A provides a table of model inputs for many locations. The lookup table values by climate zone were created by tagging each location by climate zone and taking the median value of each climate zone. The result is what you suggest below a correlation but still some variability within each climate zone. 
image

File: HPXMLtoOpenStudio/resources/weather.rb
Comment (afontani) [reply HPXMLtoOpenStudio/resources/weather.rb:345]: Using TMY3 station data, there are some differences (mainly in cooler climates) between the L. Xing data and the annual drybulb. There is some scatter, but the $r^2$ is pretty good. I would be in favor of using `weather.calc_ground_temperatures()` with an adjustment using the equation below to determine the ground temperature based on the L. Xing Model.

This way we use an established method already implemented in OS-HPXML, only make a small adjustment to the outputs of the method, and ground temperatures change for every location. 

**Drawback:** Different weather files will produce different ground temperatures for a given location (example: an AMY simulation for 2019 and 2020). This might not be bad for foundation heat transfer and shallow GSHP systems but is a little strange for vertical borefields.

drybulb-ground-temperature-correlation

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (afontani) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb:2683]: @jmaguire1 : any thoughts here. I think you compared our sizing to various established L1-L4 methods.

File: HPXMLtoOpenStudio/tests/test_weather.rb
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/tests/test_weather.rb:262]: @afontani @shorowit To be clear, we're not going to be using the new `ground_temperatures.csv` lookup file, correct? Instead, we'll use `weather.calc_ground_temperatures` (C) and then apply a small adjustment based on `y = 0.91x + 1.82` where `x = average(weather.calc_ground_temperatures)`.

Or are we _also_ updating `@data.GroundMonthlyTemps << ...` to use the lookup value instead of `data.AnnualAvgDryBulb`?

File: HPXMLtoOpenStudio/resources/weather.rb
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/weather.rb:311]: If we end up sticking with this, we should add a comment explaining where this equation/regression came from.

File: HPXMLtoOpenStudio/tests/test_weather.rb
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/tests/test_weather.rb]: Done in https://github.com/NREL/OpenStudio-HPXML/pull/1475/commits/75e61c7ff5b5da2bc4f219321c8d23d184568318.

File: HPXMLtoOpenStudio/resources/weather.rb
Comment (afontani) [reply HPXMLtoOpenStudio/resources/weather.rb:312]: @shorowit : I added a comment in [d3eb3c4](https://github.com/NREL/OpenStudio-HPXML/pull/1475/commits/d3eb3c47860e8b45532d5ea5530f9e39570a44e8)

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (shorowit) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: Let's resolve this question before merging -- do we want to make a change to address this now or not?

Comment (afontani) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: I propose we use the `weather.data.GroundMonthlyTemps` array for the January and July values for `ground_temp_f` and then call this done for now. I will create an issue about how the L. Xing model could be used to update the `weather.data.GroundMonthlyTemps` array.

Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: Denver looks like this:
```
0 | 45.59726
1 | 42.4099
2 | 42.15465
3 | 43.66721
4 | 49.73931
5 | 55.78311
6 | 61.02277
7 | 64.34197
8 | 64.62395
9 | 61.92434
10 | 56.8183
11 | 50.95989
```
So use temperatures corresponding to 0 and 6? And not min and max? (Min and max might also be easier for southern hemisphere.)



</body>

</html>

Comment (shorowit) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: I think January/July would be better than min/max, since the design loads will be calculated using the months with the hottest/coldest outdoor air temperatures. You want to know how much borehole depth you need _coincident with when those design loads occur_.

Make sure to handle locations in the southern hemisphere correctly too.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (joseph-robertson) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:2679]: Are the 0 and 6 here specific to northern hemisphere? Should we use the same logic here (i.e., switch the indexes based on latitude) as we will with the ground temperatures?

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (jmaguire1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: I'm on board with this (and I agree with Scott that Jan/July might be better than min/max, since that's consistent with a lot of our other design calculations).

@afontani: Do you remember what depth you used as an input to develop your correlation? The [Xing model](https://bigladdersoftware.com/epx/docs/8-7/engineering-reference/undisturbed-ground-temperature-model-xing.html) is a function of time and depth. I'm a little surprised how much of a seasonal swing we're seeing in the ground temperature here since we're generally talking about the temperature > 100 ft underground, I would have expected it to be a little more stable than what Joe is showing.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: I'll take that approach for now. Let me know if it's incorrect.

Comment (jmaguire1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: What you're doing here is correct

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (afontani) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:2678]: Thank you for this addition.

**Review by afontani — APPROVED** [review]
Thank you for adding a unit test.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (shorowit) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: Nice catch here, @joseph-robertson.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb
Comment (shorowit) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: Based on these changes, total bore lengths are increasing by 25-50% in our tests. That's a lot.

image

@jmaguire1's comment makes me reconsider here. Indeed our monthly ground temperatures represent shallow temperatures primarily used for foundation heat transfer modeling. If we should be using deep ground temperatures here, then using the annual average is probably (?) better than what we've changed to, though we could look at the Xing data to confirm. (If we could develop monthly ground temperatures for deeper depths, that'd be the best possible outcome of course, but that would take more work.)

File: HPXMLtoOpenStudio/resources/weather.rb
Comment (joseph-robertson) [inline HPXMLtoOpenStudio/resources/weather.rb:316]: @afontani and I plan to:

Leave this. But in `hvac_sizing.rb`, use `adj_annual_avg_drybulb_f` instead of the average of `@data.GroundMonthlyTemps`.

Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/weather.rb:321]: image

**Review by shorowit — APPROVED** [review]
Did some minor refactoring/code cleanup. Looks good to me. Will merge if the CI is still happy and results change as expected.

**Pull request merged**
